mapalign/dist.py

```diff
@@ -147,8 +148,8 @@ def compute_affinity(X, method='markov', eps=None, metric='euclidean'):
     import numpy as np
     D = squareform(pdist(X, metric=metric))
     if eps is None:
-        k = int(max(2, np.round(D.shape[0] * 0.01)))
-        eps = 2 * np.median(np.sort(D, axis=0)[k+1, :])**2
+        k = int(max(2, np.round(D.shape[0] * f_neighbors)))
+        eps = np.median(np.sort(D, axis=0)[0:k+1, :]**2)
```

Comment:
if i remember this correctly, `eps` is based on the `k+1` th entry rather than than everything upto that point. it basically provides a scaling factor given the sorted distance.

Reply:
The approach currently implemented is more liberal than the one I propose and both of them are correct. All manifold learning approaches are based on local similarities and these should be emphasized - I think the median of all the values up to the k-th captures this property more accurately. Besides, I think that parametrization of the kernel in terms of average distances to a certain fraction of nearest neighbors is more natural, because it doesn't depend on the scale of the affinities that one uses.